### ISSUE TITLE
Pass critical MicroProfile TCK tests

### ISSUE THREAD

**Issue by paulrobinson** [quarkusio/quarkus#4611]
- [x] [AirlinesAppTest](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-openapi/pom.xml#L38) from #4519 
- [x] Metrics TCK (#4518) waiting on a config update
- [x] Metrics TCK excluded tests (#5014)
- [x] Context Propagation TCK (#4517). Currently waiting on solution to #3878 
- [x] [ArrayConverterTest.*](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L17) from #4516
- [x] [ConfigProviderTest.testInjectedConfigSerializable](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L26) from #4516
- [x] [ConfigProviderTest.testPropertyConfigSource](https://github.com/quarkusio/quarkus/blob/master/tcks/microprofile-config/tck-suite.xml#L28) from #4516
- [x] Rest Client TCK (#4520) There's a lot of skipped and errored tests that need work

**Comment by paulrobinson** [comment]
@kenfinnigan Do you have a list off GH issues for the failing tests that we need to pass for Quarkus 1.0.0?

**Comment by kenfinnigan** [comment]
https://github.com/quarkusio/quarkus/issues/4520, https://github.com/quarkusio/quarkus/issues/4519, https://github.com/quarkusio/quarkus/issues/4518, https://github.com/quarkusio/quarkus/issues/4517, https://github.com/quarkusio/quarkus/issues/4516 contain the currently excluded TCK tests.

But there hasn't been any evaluation done as to what is "critical" and what isn't

**Comment by kenfinnigan** [comment]
FT we're still waiting on for a list of things to fix, as the TCK isn't part of CI yet

**Comment by paulrobinson** [comment]
@kenfinnigan who are you waiting on for this list?

**Comment by kenfinnigan** [comment]
No person, we were waiting on an MP FT release, but we can now do a SmallRye release and see how many tests we've got left in the PR

**Comment by kenfinnigan** [comment]
I've updated the epic with details of what I think is critical

**Comment by jmartisk** [comment]
I think we should add https://github.com/quarkusio/quarkus/issues/5014 now that the test was recently excluded and is quite a significant part of the Metrics REST TCK.

**Comment by emmanuelbernard** [comment]
Do you mind adding some updates @jmartisk and @kenfinnigan especially in the context of 1.1

**Comment by kenfinnigan** [comment]
What kind of update are you looking for?

We're waiting on the mega config PR to be merged, which will fix a lot of the above and I have another PR waiting on the config one being merged.

Once config PR is merged we're probably less than a week from MP TCK compatibility on everything

**Comment by kenfinnigan** [comment]
All resolved now
